packages/valory/skills/termination_abci/behaviours.py

```diff
@@ -398,10 +403,13 @@ def _get_multisend_tx(self) -> Generator[None, None, Optional[str]]:
         safe_owners = yield from self._get_safe_owners()
         if safe_owners is None:
             return None
-
         owner_to_be_swapped = safe_owners[0]
         # we remove all but one safe owner
-        for owner in safe_owners[1:]:
+        # reverse the list to avoid errors when removing owners
+        # this is because the owners are stored as a linked list
+        # in the safe, hence the order is important
+        safe_owners = list(reversed(safe_owners[1:]))
+        for owner in safe_owners:
```

Comment:
`collections.deque` is a pretty versatile data structure. It can serve as a stack, queue and as a linked list.

Reply:
Since you are getting an `iterator` and immediately converting it to a `list`, it could also have been:

```python
safe_owners = safe_owners[:0:-1]
```

Reply:
On a future release we will get rid of this reversal completely.

Reply:
you mean you're going to convert the linked list into a regular array / stack right? that is a good idea, you might want to open an issue on this or a trello ticket so you do not forget - unless you intend to do it straight away of course.

Reply:
We don't need to have a linked list, this reversal is a result of how we currently set [`prevOwner`](https://github.com/safe-global/safe-contracts/blob/v1.3.0/contracts/base/OwnerManager.sol#L71). When I did this fix it came across my mind, but I went with this because it was simpler.